docs/RELEASING.md

```diff
@@ -0,0 +1,38 @@
+# Releasing the Connectors project
+
+The Connectors project is packaged as a Gem that is added in the Enterprise Search distribution.
+
+The version scheme we use is **MAJOR.MINOR.PATCH.BUILD** and stored in the `VERSION` file 
+at the root of this repository.
+
+## Unified release
+
+**MAJOR.MINOR.PATCH** should match the Elastic and Enterprise Search version it targets 
+and the *BUILD* number should be set to **0** the day the release is created
+to be included with the Enterprise Search distribution.
+
+For example, when shipping for `8.1.2`, the version is `8.1.2.0`.
+
+To release Connectors:
+
+- set the VERSION file to the right version
+- make sure all tests and linter pass with `make lint test`
+- run `make release`
```

Comment:
We probably need additional steps to:
- tag the commit in git
- push the tag to `origin`
- if this is a new MINOR:
  - create a new maintenance branch 
  - update the VERSION on `main`
- create a Github release (idk if this is necessary?)
  - upload the gem to Github's release page (so that we have a source of truth on what the released artifact is)
- immediately update ent-search's gem dependency with the released version

Reply:
thanks, great additions, I forgot about tagging. I guess that can be added in `make release` 
+1 for the branching

For GH releases, I don't know, I have never done this -- It sounds like a good idea though, or in the Gem artifact?

Reply:
> For GH releases, I don't know, I have never done this -- It sounds like a good idea though

I hadn't done it before the other week. After the tag was there, it was just a few clicks in here: https://github.com/elastic/connectors/releases. Easy to pick which tag, if the release is a "pre-release" or not, and if you want to add release notes. And then you can upload artifacts (like the gem)

> or in the Gem artifactory service?

This would be my preference, but IDK if we have access to publish to our Elastic Artifactory, or how much work it would be to automate this. In my ideal world, we'd have a Jenkins job that ran the `make release`, created our tags/branches, and published our artifact all automatically.

Reply:
> This would be my preference, but IDK if we have access to publish to our Elastic Artifactory,

We do, but we could also publish to RubyGems and store the creds in Vault

Reply:
Ok I changed the code so we push it to RubyGems. I have created an account for ent-search-dev@elastic.co but I did not get the email back yet for registration :(